paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/changelog/ChangelogCompactTask.java

```diff
@@ -247,6 +259,39 @@ public String toString() {
                 partition, newFileChangelogFiles, compactChangelogFiles);
     }
 
+    private static class ReadTask {
+
+        private final FileStoreTable table;
+        private final Path path;
+        private final int bucket;
+        private final boolean isCompactResult;
+        private final DataFileMeta meta;
+
+        private byte[] result = null;
+
+        private ReadTask(
+                FileStoreTable table,
+                Path path,
+                int bucket,
+                boolean isCompactResult,
+                DataFileMeta meta) {
+            this.table = table;
+            this.path = path;
+            this.bucket = bucket;
+            this.isCompactResult = isCompactResult;
+            this.meta = meta;
+        }
+
+        private void readFully() {
+            try {
+                result = IOUtils.readFully(table.fileIO().newInputStream(path), true);
+                table.fileIO().deleteQuietly(path);
```

Comment:
If job failover after `table.fileIO().deleteQuietly(path);` and before `copy all files into a new big file`. 
Is there a risk of file loss here？

Reply:
If job fails then no changelog will be committed, thus no risk of file loss.